dts/bindings/adc/gd,gd32-adc.yaml

```diff
@@ -0,0 +1,54 @@
+# Copyright (c) 2021 BrainCo Inc.
+# SPDX-License-Identifier: Apache-2.0
+
+description: GigaDevice GD32 ADC
+
+compatible: "gd,gd32-adc"
+
+include: [adc-controller.yaml, pinctrl-device.yaml]
+
+properties:
+    reg:
+      required: true
+
+    rcu-periph-clock:
+      type: int
+      description: Reset Control Unit Peripheral Clock ID
+      required: true
+
+    rcu-clock-source:
+      type: int
+      description: ADC clock source and prescaler selector
+      required: false
+      default: 4
+      enum:
+        - 0 # RCU_ADCCK_IRC28M_DIV
+        - 1 # RCU_ADCCK_IRC28M
+        - 2 # RCU_ADCCK_APB2_DIV2
+        - 3 # RCU_ADCCK_AHB_DIV3
+        - 4 # RCU_ADCCK_APB2_DIV4
+        - 5 # RCU_ADCCK_AHB_DIV5
+        - 6 # RCU_ADCCK_APB2_DIV6
+        - 7 # RCU_ADCCK_AHB_DIV7
+        - 8 # RCU_ADCCK_APB2_DIV8
+        - 9 # RCU_ADCCK_AHB_DIV9
```

Comment:
Is this range 0-9 valid for all gd32 SoCs?
if yes, all comments are valid for all gd32 SoCs? I mean, value 1 always be IRC28M? Can any other enum position assume other value?

Reply:
Refer from https://github.com/zephyrproject-rtos/zephyr/pull/41855#issuecomment-1019416536, this property currently only valid for gd32f3x0 board.

rcu adc clock enum definition can be found in [rcu_adc_clock_enum](https://github.com/zephyrproject-rtos/hal_gigadevice/blob/cc85acbd635f02122c54df11fa55458269ce1bdd/gd32f3x0/standard_peripheral/include/gd32f3x0_rcu.h#L441)

Reply:
Update the property description, support all external clock source.